app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -850,4 +948,44 @@ object GoogleMapHelper {
         val rotatedCartesian = computeRotation(pointCartesian, angle)
         return inverseEquirectangularProjection(rotatedCartesian, center)
     }
+
+    /**
+     * Color all the areas of a zone to a certain color
+     * @param idZone id of the zone to color
+     * @param color color target for the zone
+     */
+    fun colorAreas(idZone: Int, color: Int) {
+        for (key in zonesToArea[idZone]!!.second) {
+            areasPoints[key]!!.third.strokeColor = color
+        }
+    }
+
+    /**
+     * clears the color of the current selected zone
+     */
+    fun clearSelectedZone() {
+        if (selectedZone != -1) {
+            colorAreas(selectedZone, DEFAULT_ZONE_STROKE_COLOR)
+            selectedZone = -1
+        }
+    }
+
+    /**
+     * Set the selected zone to color it on the map
+     * @param tag id of the selected zone
+     */
+    fun setSelectedZones(tag: Int) {
+        clearSelectedZone()
+        selectedZone = tag
+        colorAreas(tag, SELECTED_ZONE_STROKE_COLOR)
+    }
+
+    /**
+     * Set the selected zone from to color it on the map from the id of the area
+     * @param tag id of the area to find the zone it belongs
+     */
+    fun setSelectedZoneFromArea(tag: String) {
+        setSelectedZones(areasPoints[tag.toInt()]!!.first)
+    }
```

Comment:
We should seriously consider a refactor of this class since it has almost 1000 lines now

Reply:
Yes we should

Reply:
Yes maybe add a TODO on top of this class to remember refactoring it.